decompilers/d2d_binja/server.py

```diff
@@ -102,10 +102,20 @@ def function_data(self, addr: int):
 
         func = funcs[0]
 
+        # get stack frame offset for x86
+        frame_offset = 0
+        if self.bv.arch.name == 'x86_64':
+            frame_offset -= self.bv.arch.address_size
+        elif self.bv.arch.name == 'x86':
+            # handle inconsistent stack frame offsets
+            current_frame = func.get_reg_value_at(addr, 'ebp')
+            if current_frame.type != EntryRegisterValue.type:
+                frame_offset = current_frame.value
+
         # get stack vars
         stack_vars = {}
         for stack_var in func.stack_layout:
-            offset = abs(stack_var.storage)
+            offset = frame_offset - stack_var.storage
```

Comment:
@justalmill would you be able to test this when it is still wrapped in abs? Since we need to make sure the offset sent is not negative when it makes to the gdb client.

Reply:
I did some testing and the negative offsets work fine for me in gdb. 

Bizarrely enough, this is valid, and the same as just subtracting:
```
(gdb) set $test = (int*) $fp + -8
(gdb) x/gx $test
0x7fffffffddc8:	0x00007ffff7f944c0

(gdb) set $test = (int*) $fp -8
(gdb) x/gx $test
0x7fffffffddc8:	0x00007ffff7f944c0
```
I do not think the offsets should be absolute valued regardless, if anything the negative values should be tossed if you want to keep everything positive.

Since the stack_vars dictionary is using the offset as a key, if you absolute value the offsets then the negative offset stack variables that come before the frame (and get iterated last) overwrite the dictionary entries for in frame stack variables with positive offsets.